test/e2e/framework.go

```diff
@@ -1521,6 +1521,11 @@ func (data *TestData) createAgnhostNodePortService(serviceName string, affinity,
 	return data.createService(serviceName, testNamespace, 8080, 8080, map[string]string{"app": "agnhost"}, affinity, nodeLocalExternal, corev1.ServiceTypeNodePort, ipFamily)
 }
 
+// createNginxNodePortService creates a NodePort nginx service with the given name.
+func (data *TestData) createNginxNodePortService(serviceName string, affinity, nodeLocalExternal bool, ipFamily *corev1.IPFamily) (*corev1.Service, error) {
```

Comment:
Why it doesn't use agnhost?

Reply:
I need an agnhost host network Pod as client to fake an external client. If I create an agnhost NodePort, then the Service selects all agnhost as Endpoint, including the host network agnhost Pod.